modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/AddFormInstanceRecordMVCActionCommand.java

```diff
@@ -150,20 +150,24 @@ protected void doProcessAction(
 			actionRequest, "redirect", ddmFormInstanceSettings.redirectURL());
 
 		if (Validator.isNotNull(redirectURL)) {
-			portletSession.setAttribute(
-				DDMFormWebKeys.DYNAMIC_DATA_MAPPING_FORM_INSTANCE_ID,
-				formInstanceId);
-			portletSession.setAttribute(DDMFormWebKeys.GROUP_ID, groupId);
+			String currentURL = ParamUtil.getString(
+				actionRequest, "currentURL");
 
-			sendRedirect(actionRequest, actionResponse, redirectURL);
-		}
-		else {
 			DDMFormSuccessPageSettings ddmFormSuccessPageSettings =
 				ddmForm.getDDMFormSuccessPageSettings();
 
-			if (ddmFormSuccessPageSettings.isEnabled()) {
+			if (!StringUtil.equals(currentURL, redirectURL) ||
+				ddmFormSuccessPageSettings.isEnabled()) {
+
 				hideDefaultSuccessMessage(actionRequest);
 			}
+
+			portletSession.setAttribute(
+				DDMFormWebKeys.DYNAMIC_DATA_MAPPING_FORM_INSTANCE_ID,
+				formInstanceId);
+			portletSession.setAttribute(DDMFormWebKeys.GROUP_ID, groupId);
+
+			sendRedirect(actionRequest, actionResponse, redirectURL);
```

Comment:
Hi @rijema,

I noticed you merged the `if/else`, but do we really need this extra logic in the `if` statement? I understand that, after your tests, you didn't find a use case where the `redirectURL` value was empty, right?

Wouldn't it be the case of deleting the `else` instead of merging with the `if`, since it was a code that wasn't executing before?

What do we really need here to fix the reported issue?

Reply:
Hello @carolmariaabb!

This pull request was discussed on the PTR https://issues.liferay.com/browse/PTR-3312, where I would like to update the use cases for successful form submission messages.

First, I tried to update the branch with a clearer commit of my goal and also with some suggestions discussed here. Please see the new unique commit.

Goal:

According to the implementation made, I try to remove the green alert (ToastAlertContainer) from submissions that contain a redirect link and also success pages, based on the PTR discussion.

Answering your questions:

> I noticed you merged the if/else, but do we really need this extra logic in the if statement? I understand that, after your tests, you didn't find a use case where the redirectURL value was empty, right?

That's correct. RedirectURL is never empty.

> Wouldn't it be the case of deleting the else instead of merging with the if, since it was a code that wasn't executing before?

Well, the initial proposal was to take this else as a code block to be evaluated before the if, but after the first iterations in PR, I brought it into the if. That is, the content of the else is within the scope that I created for the solution of this fix, since pages that have a success page could also omit the Green Toast Alert, avoiding duplication of information in the posted content (Alert + SuccessPage).

> What do we really need here to fix the reported issue?

Basically, prevent that when returning to the form, after being redirected to another page, and the notification message from being launched.

Reply:
Hi @rijema,

If I understood correctly, after your changes, there is no more green alert when Forms is using the **default** success page and **not** the Redirect URL on Success, right? 

But this scenario was not discussed on the PTR,

> In addition, according to the current implementation, when the setting Redirect URL on Success is configured then the Success Page and the success notification won't be displayed, because Forms understands that the redirected URL will provide the necessary success information to the user who just submitted the form."

The comment above is only mentioning the use case where Redirect URL on Success is configured.

Could you confirm in the PTR the expected behavior when using the default success page?

Also, I believe that some Poshi tests will fail, as we are expecting the green alert to appear when the form is submitted using the default success page.

Thanks!

Reply:
Of course, @carolmariaabb . Thanks for your entries. I concluded that because the code was understanding the default success page as a redirected URL, I included it in the restriction. I'll ask one more question on the PTR. Thanks in advance!

Reply:
According to the last team comment on https://issues.liferay.com/browse/PTR-3312, if success page is defined, success green alert should be on the page. Pull request updated. Regards!